plugins/node/opentelemetry-instrumentation-graphql/src/utils.ts

```diff
@@ -32,6 +32,32 @@ import {
 
 const OPERATION_VALUES = Object.values(AllowedOperationTypes);
 
+function addInputVariableAttribute(span: api.Span, key: string, variable: any) {
+  switch (typeof variable) {
```

Comment:
I think this would be enough 

```
      if (Array.isArray(variable)) {
        variable.forEach((value, idx) => {
          addInputVariableAttribute(span, `${key}.${idx}`, value);
        });
      } else if (variable instanceof Object) {
        Object.entries(variable).forEach(([nestedKey, value]) => {
          addInputVariableAttribute(span, `${key}.${nestedKey}`, value);
        });
      }
      else {
        span.setAttribute(`${AttributeNames.VARIABLES}${String(key)}`, variable);
      }
```

Reply:
Fixed